sx127x/sx127x.go

```diff
@@ -506,26 +523,35 @@ func (d *Device) RandomU32() uint32 {
 
 // HandleInterrupt must be called by main code on DIO state change.
 func (d *Device) HandleInterrupt() {
+
 	// Get IRQ and clear
 	st := d.ReadRegister(SX127X_REG_IRQ_FLAGS)
 	d.WriteRegister(SX127X_REG_IRQ_FLAGS, 0xFF)
 
-	rChan := d.GetRadioEventChan()
-
 	if (st & SX127X_IRQ_LORA_RXDONE_MASK) > 0 {
-		rChan <- lora.NewRadioEvent(lora.RadioEventRxDone, uint16(st), nil)
+		e := lora.RadioEvent{lora.RadioEventRxDone, uint16(st), nil}
+		d.radioEventChan <- e
 	}
 
 	if (st & SX127X_IRQ_LORA_TXDONE_MASK) > 0 {
-		rChan <- lora.NewRadioEvent(lora.RadioEventTxDone, uint16(st), nil)
+		e := lora.RadioEvent{lora.RadioEventTxDone, uint16(st), nil}
+		d.radioEventChan <- e
 	}
 
 	if (st & SX127X_IRQ_LORA_RXTOUT_MASK) > 0 {
-		rChan <- lora.NewRadioEvent(lora.RadioEventTimeout, uint16(st), nil)
+		e := lora.RadioEvent{lora.RadioEventTimeout, uint16(st), nil}
+		select {
+		case d.radioEventChan <- e:
```

Comment:
Can do this in one line like the above, I think

Reply:
Fixed